dds/DCPS/ThreadPool.h

```diff
@@ -0,0 +1,55 @@
+/*
+ *
+ *
+ * Distributed under the OpenDDS License.
+ * See: http://www.opendds.org/license.html
+ */
+
+#ifndef OPENDDS_DCPS_THREAD_POOL_H
+#define OPENDDS_DCPS_THREAD_POOL_H
+
+#include "dcps_export.h"
+
+#include "Barrier.h"
+#include "PoolAllocator.h"
+
+#include <ace/Thread.h>
+
+OPENDDS_BEGIN_VERSIONED_NAMESPACE_DECL
+
+namespace OpenDDS
+{
+namespace DCPS
+{
+
+class OpenDDS_Dcps_Export ThreadPool
+{
+public:
+
+  typedef ACE_THR_FUNC_RETURN(*FunPtr)(void*);
+
+  ThreadPool(size_t count, FunPtr fun, void* arg = 0);
+  virtual ~ThreadPool();
+
+  static ACE_THR_FUNC_RETURN run(void* arg);
```

Comment:
This seems much simpler / cleaner to me, since it doesn't pull in ACE_Service_Object / ACE_Event_Handler / ACE_Shared_Object. As it stands, this could pretty easily be migrated (or #ifdef'd) to C++11 only if we switch to std::thread.